### ISSUE TITLE
Generic Install missing release zip

### ISSUE THREAD

**Issue by cuhde** [45Drives/cockpit-file-sharing#124]
The release page is missing the generic zip files for a lot of versions. The current version 4.2.8 has instructions to download the generic zip 
"curl -LO https://github.com/45Drives/cockpit-file-sharing/releases/download/v4.2.8/cockpit-file-sharing_4.2.8_generic.zip"
which does not exist in the release.

**Comment by joshuaboud** [comment]
I'll need to update the generic installation instructions...
The generic zip file method was used a few releases back due to being built against a private NPM repo, but now that all of the dependencies are public (the [houston-common submodule](https://github.com/45Drives/houston-common)), you should be able to just install from source.

Try these steps:
1. `git clone https://github.com/45Drives/cockpit-file-sharing.git`
2. `cd cockpit-file-sharing`
3. `make`
4. `sudo make install` (installs to /usr/share/cockpit) OR `make install-local` (as normal user, installs to ~/.local/share/cockpit)
5. `sudo systemctl restart cockpit` may be needed if installed to /usr/share/cockpit

**Comment by svandive** [comment]
Sorry this isn't working for me.  Are their dependencies (like yarn) that need to be installed first?  What I am seeing for output from make right now it:

```
root@homebridge:~/git/cockpit-file-sharing# make
./bootstrap.sh
+ command -v sponge
+ jq 'del(.packageManager)' ./package.json
+ sponge ./package.json
+ rm .yarnrc.yml .yarn -rf
+ yarn set version stable
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'set'
make: *** [Makefile:73: .yarnrc.yml] Error 1
```

**Comment by mikedebian** [comment]
Yeah, clone the houston library that was linked, make in it and copy the directory to the filesharing git directory and continue the steps.

**Comment by TheFlagCourier** [comment]
Some notes from setting this up on AlmaLinux 9

## Dependencies

There are three packages that are not listed as dependencies, but are required to build with. You'll need to enable the [EPEL](https://docs.fedoraproject.org/en-US/epel/) repository to use them.

- Node.js (>= 18.12). Use a module, not the regular package.
- yarnpkg
- moreutils

## Working with the Submodule

You will need a valid SSH key associated with a GitHub account to pull the submodule. Alternatively, edit the origin to use HTTPS in the git config.

## Helper Script

```bash
#! /usr/bin/env bash

## Initialize submodule
git clone --recurse-submodules https://github.com/45Drives/cockpit-file-sharing

# If already cloned: #
# cd cockpit-file-sharing
# git submodule init

# Set up EPEL
sudo dnf install epel-release
sudo dnf config-manager --set-enabled crb

# yarnpkg will pull in Node.js 16 as a regular package.
sudo dnf module install nodejs:22

# Required packages (in addition to packages on readme)
sudo dnf install moreutils yarnpkg
```

**Comment by joshuaboud** [comment]
Thanks for the heads up on the submodule requiring SSH. Instead of hard coding to HTTPS I followed [this guide to use a relative submodule URL](https://www.damirscorner.com/blog/posts/20210423-ChangingUrlsOfGitSubmodules.html). Not pushed to main yet but it is in [4.2.9-staging](/45Drives/cockpit-file-sharing/tree/4.2.9-staging)

**Comment by swilkens** [comment]
Coming from Fedora Server 41 - generic installation instructions resulted in unhappy Cockpit.

The last release to actually have a -generic appears to be [Cockpit File Sharing 4.0.2-1](https://github.com/45Drives/cockpit-file-sharing/releases/tag/v4.0.2)
